src/axom/sidre/docs/sphinx/query_data.rst

```diff
@@ -0,0 +1,101 @@
+.. ## Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _dataconcepts-label:
+
+===========================
+Datastore Queries
+===========================
+
+Sidre provides a couple of methods that can be called to query certain
+information about the contents of a datastore object. Each method takes
+a Conduit Node and inserts information related to the query into it. The
+caller can then inspect the information by retrieving values of specifically 
+named fields from the Node. This is often helpful for debugging.
+
+-------------
+Buffer Query
+-------------
+
+A datastore object can be queried for information about the buffer objects it
+contains. For example::
+
+  Datastore* ds = ...;
+
+  conduit::Node n;
+  ds->getBufferInfo(n);
+  n.print();   // print Conduit node contents to std out, if desired
```

Comment:
Presumably, when using MPI, each rank will independently dump its info to stdout.

Should we mention that in the MPI case, it can be convenient to dump the data in separate files whose names contain the rank ID?

Reply:
I added the print method call based on @white238 comment. At this level, it is up to the user to choose what to do if they want to print the conduit Node contents.  I will look through the conduit docs and make a pointer to that if there is something about the distributed memory case. If not, I will add a note here.